portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -65,38 +69,46 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   }
 
   /**
-   * Process user comments and store them into Datastore
+   * Process user comments and email and proceeds to store both items into Datastore
    * @param request the request message
    * @param response the response message
    * @throws IOException if an I/O error occurs
    */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {    
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    UserService userService = UserServiceFactory.getUserService();
+
+    // Validate that user is still logged in before adding their input to Datastore
+    if (userService.isUserLoggedIn()) {
+        String userEmail = userService.getCurrentUser().getEmail();
+
+        // Get the input from the form.
+        ArrayList<String> comments = new ArrayList<String>();
+        String userComments = getParameter(request, "text-input", "");
+        String[] words = userComments.split("\\s*,\\s*");
+        
+        // Store each comment as an entity in Datastore
+        for (String comment : words) {
+            long timestamp = System.currentTimeMillis();
+            Entity commentEntity = new Entity("Comment");
+            commentEntity.setProperty("email", userEmail);
+            commentEntity.setProperty("text", comment);
+            commentEntity.setProperty("timestamp", timestamp);        
+
+            datastore.put(commentEntity);
+        }
 
-    // Get the input from the form.
-    ArrayList<String> comments = new ArrayList<String>();
-    String userComments = getParameter(request, "text-input", "");
-    String[] words = userComments.split("\\s*,\\s*");
-    
-    // Store each comment as an entity in Datastore
-    for (String comment : words) {
-        long timestamp = System.currentTimeMillis();
-        Entity commentEntity = new Entity("Comment");
-        commentEntity.setProperty("text", comment);
-        commentEntity.setProperty("timestamp", timestamp);        
-
-        datastore.put(commentEntity);
     }
 
     response.sendRedirect("/index.html");
   }
   
   /**
-   * Returns String input entered by the commenter
-   * @return the user's input, or the default data for an input element
-   * @param name of the input element or text area to retrieve user input from 
-   * @param defaultValue to return if the user doesn't enter their own input
+    * Returns String input entered by the commenter
+    * @return the user's input, or the default data for an input element
+    * @param name of the input element or text area to retrieve user input from 
+    * @param defaultValue to return if the user doesn't enter their own input
```

Comment:
I think the spacing was correct before
(hope i didn't give misleading advice before!)
```
/**
 * Method summary
 * @any params or returns or throws or whatever
 */
```
while we're here tho, i think the return comes last in the list, after the params

Reply:
noted, thank you!